### PR TITLE
Update for more recent Julia versions

### PR DESCRIPTION
**Pull request by rdeits**
* Require Julia 1.6 (the current LTS) as the minimum Julia version
* Require Rotations.jl 1.1 and rename UnitQuaternion -> QuatRotation to match
* Update all Manifest.toml files to the new format and re-generate them on the latest Julia release
* Test Julia 1.6 and the latest release on CI

### CODE REVIEW

**Comment by ferrolho** [comment]
Hi Robin! Thanks for taking the time to do this.

FYI, I have an open MR for the rotations: #629. It looks very similar to your changes, except perhaps for
```julia
quat = convert(QuatRotation{T}, rot)
w, x, y, z = Rotations.params(quat)
```
which I remember having to do at that time to retrieve `w, x, y, z`; but maybe now we can get it without doing `Rotations.params(quat)`.

On a related note, I've created a new release on my personal fork just yesterday: [ferrolho/RigidBodyDynamics.jl@v2.4.0](https://github.com/ferrolho/RigidBodyDynamics.jl/releases/tag/v2.4.0). Maybe it can serve as inspiration for what the next release of the Julia Robotics repo could be. I'm happy to help taking this through!

**Comment by rdeits** [comment]
Ah, nice, I prefer your `params` solution over directly accessing the fields. I've cherry-picked your commits into this PR. Thanks for the updated release--once we have all the tests and CI working here we can create another one with the fast version of Rotations.jl.

**Comment by ferrolho** [comment]
Remaining CI error can be fixed with #636

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "RigidBodyDynamics"
 uuid = "366cf18f-59d5-5db9-a4de-86a9f6786172"
-version = "2.3.2"
+version = "2.5.0"
```
Comment (ferrolho) [inline Project.toml:3]: We could increment it to 2.4.0 instead.

**Review by ferrolho — APPROVED** [review]

File: Project.toml (hunk 1 above)
Comment (rdeits) [reply Project.toml]: fixed, thanks

Comment (ferrolho) [reply Project.toml]: Good to merge from my end (I don't have permissions to do it).

**Pull request merged**